pkg/sql/sem/builtins/builtins.go

```diff
@@ -9629,3 +9736,26 @@ func prettyStatement(p tree.PrettyCfg, stmt string) (string, error) {
 	}
 	return formattedStmt.String(), nil
 }
+
+func floatToTimestampTZ(ts float64) (tree.Datum, error) {
+	if math.IsNaN(ts) {
+		return nil, pgerror.New(pgcode.DatetimeFieldOverflow, "timestamp cannot be NaN")
+	}
+	if ts == math.Inf(0) {
```

Comment:
this should be Inf(1) - please add a test for pos/neg inf.

Reply:
Inf(0) and Inf(1) are the same thing, but just fixed